plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java

```diff
@@ -550,12 +582,11 @@ private void indexRemoteModel(
                     .tenantId(registerModelInput.getTenantId())
                     .build();
 
-                IndexRequest indexModelMetaRequest = new IndexRequest(ML_MODEL_INDEX);
-                if (registerModelInput.getIsHidden() != null && registerModelInput.getIsHidden()) {
-                    indexModelMetaRequest.id(modelName);
-                }
-                indexModelMetaRequest.source(mlModelMeta.toXContent(XContentBuilder.builder(JSON.xContent()), EMPTY_PARAMS));
-                indexModelMetaRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+                PutDataObjectRequest putModelMetaRequest = new PutDataObjectRequest.Builder()
+                    .index(ML_MODEL_INDEX)
+                    .id(Boolean.TRUE.equals(registerModelInput.getIsHidden()) ? modelName : null)
```

Comment:
in this case model id will be null or auto generated in the index?

Reply:
Yes, if `id` is omitted in an index request it auto-generates a UUID.

This was an attempt to faithfully reproduce this.  Yoda expressions avoid the need for the null check and are much more readable!
```java
                if (registerModelInput.getIsHidden() != null && registerModelInput.getIsHidden()) {
                    indexModelMetaRequest.id(modelName);
                }
```